### PR TITLE
Replace deprecated APIs for Input and Output with the new versions.

### PR DESCRIPTION
**Pull request by anwar6953**
As a part of https://issues.cask.co/browse/CDAP-5651, we added new a new API and deprecated three.
This PR changes usages of those 3 methods to the new one.
It also makes the corresponding changes for Multiple Inputs of MapReduce (https://issues.cask.co/browse/CDAP-3980)

http://builds.cask.co/browse/CDAP-DUT3996

Note: I'll make updates to the docs hashes at the end, once code review is done.

### CODE REVIEW

**Comment by awholegunch** [comment]
These documentation files are affected by this change, and need editing because the line numbers they are including need to be adjusted for the lines that are being added here:

TopNMapReduce.java 
  cdap-docs/developers-manual/source/building-blocks/workflows.rst
  cdap-docs/developers-manual/build.sh

StreamConversionMapReduce.java
  cdap-docs/examples-manual/source/examples/stream-conversion.rst

ScoreCounter.java
  cdap-docs/examples-manual/source/examples/sport-results.rst

SparkPageRankApp.java
  cdap-docs/developers-manual/source/building-blocks/spark-programs.rst
  cdap-docs/examples-manual/source/examples/spark-page-rank.rst

PurchaseHistoryBuilder.java
  cdap-docs/developers-manual/source/building-blocks/mapreduce-programs.rst
  cdap-docs/examples-manual/source/examples/purchase.rst
  cdap-docs/developers-manual/source/building-blocks/services.rst

WordCount.java
  cdap-docs/examples-manual/source/examples/word-count.rst
  cdap-docs/examples-manual/source/examples/fileset-example.rst

DataCleansingMapReduce.java
  cdap-docs/examples-manual/source/examples/data-cleansing.rst

**Comment by anwar6953** [comment]
Ok, thanks. I'll take care of docs hashes once code is LGTM.

**Comment by anwar6953** [comment]
@awholegunch Updated those files now.

File: cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceConfigurer.java [hunk 1]
```diff
@@ -48,7 +48,7 @@
   /**
    * Sets the name of the dataset used as output for the {@link MapReduce}.
    *
-   * @deprecated as of 3.4.0. Use {@link MapReduceContext#addOutput(String)}
+   * @deprecated as of 3.4.0. Use {@link MapReduceContext#addOutput(co.cask.cdap.api.data.batch.Output)}
```
Comment (anew) [inline cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceConfigurer.java]: does this have to be fully qualified?

File: cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java [hunk 2]
```diff
@@ -178,7 +178,7 @@ protected final void useStreamInput(StreamBatchReadable streamBatchReadable) {
   /**
    * Sets the name of the Dataset used as output for the {@link MapReduce}.
    *
-   * @deprecated as of 3.2.0. Use {@link MapReduceContext#addOutput(String datasetName)}
+   * @deprecated as of 3.2.0. Use {@link MapReduceContext#addOutput(co.cask.cdap.api.data.batch.Output output)}
```
Comment (anew) [inline cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java]: does this have to be fully qualified?

Comment (anwar6953) [inline cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java]: No, I'll put in the import.

**Comment by anew** [comment]
Two minor comments, other than that LGTM

**Comment by anwar6953** [comment]
Addressed comments and made a fix ("result" ->"results").
Will squash+merge once build passes.

**Pull request merged**